README.md

```diff
@@ -1,38 +1,89 @@
+
+### Extension points
+
+While remote-ui provides the component model, and takes care of propagating updates to an extension’s UI into the host application, it does not have any built-in notion of an “extension” at all. To create an extension system, Shopify authors a bit of extra code: a mapping of “extension points” (which, in code, are just strings with a specific naming format), and a way to load third party code that can register to be called for those extension points. This might sound little complicated, but the code to put it all together is actually quite straightforward. The Argo ”runtime” code, which is used to construct the sandbox your extension will run in, looks almost exactly like this for all Argo-capable surface areas in Shopify:
+
+```js
+// We keep a mapping of all the extension points you register for
+const registeredExtensionPoints = new Map();
+
+// We define a globally-available `shopify` object. The only thing this object
+// can do is register an extension (which is just a callback function) for a
+// specific extension point.
+globalThis.shopify = {
+  extend(extensionPoint, extension) {
+    registeredExtensionPoints.set(extensionPoint, extension);
+  },
+};
+
+// Here’s the function the host application will call to load your extension’s
+// script into the sandbox
+export function load(script) {
+  // Internal workings — don’t worry too much about this, because we might change
+  // it in the future as new web platform features are available!
+}
+
+// Once your script is loaded, it has had its chance to register for extension
+// points, and we can now simply call your extension with the arguments from the
+// host application, like the data your extension has access to and the UI “root“
+// your components will be attached to.
+export function run(extensionPoint, ...args) {
+  return registeredExtensionPoints.get(extensionPoint)?.(...args);
+}
+```
+
+In your extension code, you make use of these APIs using patterns that feel idiomatic in JavaScript:
+
+```js
+shopify.extend('Checkout::Feature::Render', (root) => {
+  root.appendChild(root.createText('Hello world!'));
+});
+
+// Or, if you are using our NPM libraries, we offer a bit of sugar over the
+// global:
+
+import {extend} from '@shopify/argo-checkout';
+
+extend('Checkout::Feature::Render', (root) => {
+  root.appendChild(root.createText('Hello world!'));
+});
+```
+
+As with UI components, the Argo extensibility framework lets us provide different arguments to your extension, depending the extension point you selected and other factors, like your app’s permissions. For full details on what arguments are passed for each extension point, please refer to the documentation for the [Argo-supporting surface area you want to extend](#i-just-want-to-build-an-argo-extension-not-learn-about-them).
```

Comment:
typo: ...different arguments to your extension, depending **on** the extension point...